### PR TITLE
Add activity tracking. Says it all.

### PR DESCRIPTION
**Pull request by brandoge91**
Super cool!

The user has to enable it. There is also xp per minute in game in config.ts

The roblox scripts also contain a README and comments on how to set it up.

Also adds game activity to info command, and an activity command.

### CODE REVIEW

**Comment by brandoge91** [comment]
Don't accept this yet. Later today I will be adding more commands to force add and remove activity from users (as a punishment ig???? idk. someone said i should)

File: src/commands/activity/activity.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+import {
+    getActivityEmbed,
+    getInvalidRobloxUserEmbed,
+    getRobloxUserIsNotMemberEmbed,
+    getUnexpectedErrorEmbed,
+    getUserInfoEmbed,
+} from '../../handlers/locale';
+import { provider } from '../../database/router';
+
+class ActivityCommand extends Command {
```
Comment (yogurtsyum) [inline src/commands/activity/activity.ts:16]: Why not just combine this with the info command? Having another one seems unnecessary.

File: src/commands/activity/addactivity.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+                    description: 'How many minutes do you want to add?',
+                    autocomplete: false,
+                    required: true,
+                    type: 'String',
+                }
+            ]
+        });
+    }
+
+    async run(ctx: CommandContext) {
```
Comment (yogurtsyum) [inline src/commands/activity/addactivity.ts:43]: Please copy the snippet we use to get Roblox users from another command (like promote).

And put anything that might error into a try/catch loop.

File: src/commands/activity/removeactivity.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+                    description: 'How many minutes do you want to remove?',
+                    autocomplete: false,
+                    required: true,
+                    type: 'String',
+                }
+            ]
+        });
+    }
+
+    async run(ctx: CommandContext) {
```
Comment (yogurtsyum) [inline src/commands/activity/removeactivity.ts:44]: Refer to my comment on the add activity command.

File: src/resources/commands.json [hunk 4]
```diff
@@ -1 +1 @@
...
..."INTEGER","autocomplete":false,"required":true,"choices":[],"options":[]},{"name":"reason","description":"If you would like a reason to be supplied in the logs, put it here.","type":"STRING","autocomplete":false,"required":false,"choices":[],"options":[]}],"defaultPermission":true},{"name":"removexp","description":"Removes XP from a user.","type":"CHAT_INPUT","options":[{"name":"roblox-user","description":"Who do you want to remove XP from?","type":"STRING","autocomplete":true,"required":true,"choices":[],"options":[]},{"name":"decrement","description":"How much XP would you like to remove?","type":"INTEGER","autocomplete":false,"required":true,"choices":[],"options":[]},{"name":"reason","description":"If you would like a reason to be supplied in the logs, put it here.","type":"STRING","autocomplete":false,"required":false,"choices":[],"options":[]}],"defaultPermission":true},{"name":"xprankup","description":"Ranks a user up based on their XP.","type":"CHAT_INPUT","options":[{"name":"roblox-user","description":"Who do you want to attempt to rankup? This defaults to yourself.","type":"STRING","autocomplete":true,"required":false,"choices":[],"options":[]}],"defaultPermission":true}]
```
Comment (yogurtsyum) [inline src/resources/commands.json:1]: This shouldn't have been pushed. I can remove it though.

**Review by yogurtsyum — CHANGES_REQUESTED** [review]
Looks good but I need a few changes done before pushing this.

File: src/commands/activity/addactivity.ts (hunk 2 above)
Comment (brandoge91) [reply src/commands/activity/addactivity.ts:43]: alright.

File: src/commands/activity/activity.ts (hunk 1 above)
Comment (brandoge91) [reply src/commands/activity/activity.ts:16]: okay

**Comment by brandoge91** [comment]
i am still working on this, on the quota.

**Comment by brandoge91** [comment]
recreating this PR as git broke sorry lmao

**Pull request closed without merging**